File: frontend/src/app/models/OverdueBorrowBooksDto.ts
export interface OverdueBorrowBooksDto {
    borrowId: number;
    bookId: number;
    bookTitle: string;
    isbn?: string | null;
    borrowedByUserId: number;
    borrowedByUsername: string;
    borrowedAt: string;
    dueAt?: string | null;
    daysOverdue: number;
}

File: backend/BookManagement.Api/Dtos/OverdueBorrowBooksDto.cs
namespace BookManagement.Api.Dtos
{
    public class OverdueBorrowBooksDto
    {
        public int BorrowId { get; set; }
        public int? BookId { get; set; }
        public string BookTitle { get; set; } = "";
        public string? Isbn { get; set; }
        public int BorrowedByUserId { get; set; }
        public string BorrowedByUsername { get; set; } = "";
        public DateTime BorrowedAt { get; set; }
        public DateTime? DueAt { get; set; }
        public int DaysOverdue { get; set; }
    }
}
